src/libexpr/flake/flakeref.cc

```diff
@@ -109,9 +103,14 @@ std::pair<FlakeRef, std::string> parseFlakeRefWithFragment(
             percentDecode(match.str(6)));
     }
 
-    else if (std::regex_match(url, match, pathUrlRegex)) {
-        std::string path = match[1];
-        std::string fragment = percentDecode(match.str(3));
+    else if (pathExists(url.substr(0, url.find("#")))) {
```

Comment:
Some notes (for myself mostly) on that change since it's a fairly crucial one:

- The `pathExists` call is potentially problematic because the flakeref should exist independently of what the filesystem contains. Maybe we could just split unconditionnally regardless of whether the path exist
- Splitting on `#` is probably not enough since the syntax of a path flakeref is of the form `<path>(\?<params>)?(#<fragments>)?`. So we'll want to split on `?` too
- I guess this isn't truly correct if multi-byte utf8 glyphs can contain `\63` (the ascii code for `?`) since we'd just split here. I think this isn't the case (precisely to allow this kind of things), but even if it's the case, it's probably better than the current situation

Reply:
> if multi-byte utf8 glyphs can contain \63 

I don't think they can, utf8 is carefully defined that the ascii bytes are only used for the coresponding character. All the other stuff has the highest bit set